src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/Cohost/RazorCohostDynamicRegistrationService.cs

```diff
@@ -45,16 +42,18 @@ public async Task RegisterAsync(string clientCapabilitiesString, RazorCohostRequ
             return;
         }
 
+        // TODO: Should we delay everything below this line until a Razor file is opened?
```

Comment:
This only matters for workspace endpoints right? The only one I can think of would be mapcode for now. Diagnostics can be answered through source generator

Reply:
Yes. I didn't realise mapcode was a workspace endpoint, but if so, its the only one we support. With cohosting we actually have to decide if we want to actively suppress closed file diagnostics, because we don't support it now, but we'll get it "for free" from the source generator.